plan/new_physical_plan_builder.go

```diff
@@ -110,21 +110,60 @@ func (p *LogicalJoin) convert2NewPhysicalPlan(prop *requiredProp) (taskProfile,
 	}
 	switch p.JoinType {
 	case SemiJoin, LeftOuterSemiJoin:
-		task, err = p.convert2SemiJoin()
+		task, err = p.convert2SemiJoin(prop)
 	default:
-		// TODO: We will consider smj and index look up join in the future.
-		task, err = p.convert2HashJoin()
+		if p.preferUseMergeJoin() {
+			task, err = p.convert2MergeJoin(prop)
+		} else {
+			// TODO: We will consider index look up join in the future.
+			task, err = p.convert2HashJoin(prop)
+		}
 	}
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	// Because hash join is executed by multiple goroutines, it will not propagate physical property any more.
-	// TODO: We will consider the problem of property again for parallel execution.
-	task = prop.enforceProperty(task, p.ctx, p.allocator)
 	return task, p.storeTaskProfile(prop, task)
 }
 
-func (p *LogicalJoin) convert2SemiJoin() (taskProfile, error) {
+func (p *LogicalJoin) preferUseMergeJoin() bool {
+	return p.preferMergeJoin && len(p.EqualConditions) == 1
+}
+
+func (p *LogicalJoin) convert2MergeJoin(prop *requiredProp) (taskProfile, error) {
```

Comment:
This function has the assumption that the length of EqualCondtion is one, which creates an unnecessary dependency on the implementation of `preferUseMergeJoin`.

How about rename it to `tryToConvert2MergeJoin`, check preferMergeJoin within this function?

If we can not convert it to merge join, fall back to hash join.